html2wp/forms.php

```diff
@@ -0,0 +1,408 @@
+                            1 == $confirmation["isDefault"]) {
+                                if ("message" == $confirmation["type"]) {
+                                    $response = array($confirmation["message"]);
+
+                                } else if ("page" == $confirmation["type"]) {
+                                    $uri = home_url() . "?p=" . $confirmation["pageId"];
+                                    if (!empty($confirmation["queryString"])) {
+                                        $uri .= "?" . $confirmation["queryString"];
+                                    }
+                                    wp_redirect( $uri );
+                                    exit;
+
+                                } else if ("redirect" == $confirmation["type"]) {
+                                    $uri = $confirmation["url"];
+                                    if (!empty($confirmation["queryString"])) {
+                                        $uri .= "?" . $confirmation["queryString"];
+                                    }
+                                    wp_redirect( $uri );
+                                    exit;
+                                }
+
+                                //exit the loop
+                                break;
+                            }
+                    }
+
+                    /**
+                     * Set the default form submission success message
+                     * if the response was not set from the Default Confirmations above
+                     * or if the default confirmation message was was empty.
+                     * $response[1] is the message
+                     */
+                    if (empty($response) || empty($response[1])) {
+                        $response = array("Thanks for your submission!");
+                    }
+                }
+            } else {
+                //A form was not found corresponding to the 
+                //GFForm that the user is trying to submit to
+                $response = array("Form Error");
+            }
+        } else {
+            //Nonce check failed
+            //OR gfformid is not set
+            //OR gfnonce is not set
+            $response = array("Bad Input");
+        }
+
+        //Show this if request is AJAX form submit
+        if(isset($_SERVER['HTTP_X_REQUESTED_WITH']) && strtolower($_SERVER['HTTP_X_REQUESTED_WITH']) == 'xmlhttprequest') { 
+            header('content-type: application/json; charset=utf-8');
+            echo json_encode($response)."\n";
+            exit;
+        }
+
+        //this is shown only if it is a regular form submit
+        
+        //Header
+        get_header();
+
+        //print the response message
+        echo "<h3>" . $response[0] . "</h3>";
+        
+        //Footer
+        get_footer();
+
+    }
+}
+
+//Perform setup after this theme is activated
+add_action('after_switch_theme', 'html2wp_setup_theme_components');
```

Comment:
Move html2wp_setup_theme_components and html2wp_detect_plugin_activation to setup.php and merge with existing logic if possible. Also list all actions in the beggining  of the setup.php file.